applications/plugins/SofaCarving/CarvingManager.cpp

```diff
@@ -90,15 +90,10 @@ void CarvingManager::init()
         // We look for a CollisionModel identified with the CarvingSurface Tag.
         std::vector<core::CollisionModel*> models;
         getContext()->get<core::CollisionModel>(&models, core::objectmodel::Tag("CarvingSurface"), core::objectmodel::BaseContext::SearchRoot);
```

Comment:
```suggestion
        getContext()->get<core::CollisionModel>(&m_surfaceCollisionModels, core::objectmodel::Tag("CarvingSurface"), core::objectmodel::BaseContext::SearchRoot);
```

Reply:
In fact the whole for loop is not needed. You can directly use the m_surfaceCollisionModels vector.